Source/WebCore/accessibility/AccessibilityTableRow.cpp

```diff
@@ -106,9 +108,9 @@ AccessibilityTable* AccessibilityTableRow::parentTable() const
         // If this is a non-anonymous table object, but not an accessibility table, we should stop because we don't want to
         // choose another ancestor table as this row's table.
         if (auto* parentTable = dynamicDowncast<AccessibilityTable>(*parent)) {
-            if (parentTable->isExposable())
+            if (parentTable->isExposable() && (!isARIAGridRow() || parentTable->isAriaTable() || elementName() == ElementName::HTML_tr))
```

Comment:
`(!isARIAGridRow() || parentTable->isAriaTable() || elementName() == ElementName::HTML_tr)`

Can this be extracted to a named variable? Right now this bundle of conditions is confusing